src/function/register_function_list.cpp

```diff
@@ -1,15 +1,91 @@
+#include "duckdb/catalog/default/default_types.hpp"
+#include "duckdb/common/string_util.hpp"
 #include "duckdb/function/function_list.hpp"
 #include "duckdb/parser/parsed_data/create_aggregate_function_info.hpp"
 #include "duckdb/parser/parsed_data/create_scalar_function_info.hpp"
+#include "duckdb/parser/parser.hpp"
 
 namespace duckdb {
 
+static LogicalType GetLogicalType(string &logical_type_string) {
+	LogicalType type;
+	if (StringUtil::Contains(logical_type_string, "ANY")) {
+		if (StringUtil::CIEquals(logical_type_string, "ANY")) {
+			type = LogicalType::ANY;
+		} else if (StringUtil::CIEquals(logical_type_string, "ANY[]")) {
+			type = LogicalType::LIST(LogicalType::ANY);
+		} else if (StringUtil::CIEquals(logical_type_string, "ANY[][]")) {
+			type = LogicalType::LIST(LogicalType::LIST(LogicalType::ANY));
+		} else if (StringUtil::CIEquals(logical_type_string, "MAP(ANY,ANY)")) {
+			type = LogicalType::MAP(LogicalType::ANY, LogicalType::ANY);
+		} else {
+			throw InternalException("Unsupported type: '%s'", logical_type_string);
+		}
+	} else {
+		type = Parser::ParseLogicalType(logical_type_string);
```

Comment:
Could we use `DBConfig::ParseLogicalType` here?

Reply:
done